qoqo-qryd/tests/integration/api_backend.rs

```diff
@@ -262,59 +306,292 @@ fn test_query_job_status_fail() {
-                    .call_method1("get_job_status", (job_loc,))
-                    .unwrap()
-                    .extract()
-                    .unwrap();
-                let job_status = status_report.get("status").unwrap();
-                status = job_status.clone();
-                thread::sleep(fifteen);
-
-                if status == *"completed" {
-                    assert_eq!(status, "completed");
-                    let _job_result = backend.call_method1("get_job_result", (job_loc,)).unwrap();
-                }
-            }
-        });
-    }
-}
+    let server = MockServer::start();
+    let qryd_job_status_in_progress = QRydJobStatus {
+        status: "in progress".to_string(),
+        msg: "the job is still in progress".to_string(),
+    };
+    let qryd_job_status_completed = QRydJobStatus {
+        status: "completed".to_string(),
+        msg: "the job has been completed".to_string(),
+    };
+    let result_counts = ResultCounts {
+        counts: HashMap::from([("0x1".to_string(), 100), ("0x4".to_string(), 20)]),
+    };
+    let qryd_job_result_completed = QRydJobResult {
+        data: result_counts,
+        time_taken: 0.23,
+        noise: "noise".to_string(),
+        method: "method".to_string(),
+        device: "QrydEmuSquareDevice".to_string(),
+        num_qubits: 4,
+        num_clbits: 4,
+        fusion_max_qubits: 4,
+        fusion_avg_qubits: 4.0,
+        fusion_generated_gates: 100,
+        executed_single_qubit_gates: 50,
+        executed_two_qubit_gates: 50,
+    };
+    let mock_post = server.mock(|when, then| {
+        when.method("POST");
+        then.status(201).header(
+            "Location",
+            format!("http://127.0.0.1:{}/DummyLocation", server.port()),
+        );
+    });
+    let mut mock_status0 = server.mock(|when, then| {
+        when.method("GET").path("/DummyLocation/status");
+        then.status(200).json_body_obj(&qryd_job_status_in_progress);
+    });
+    let mock_result = server.mock(|when, then| {
+        when.method("GET").path("/DummyLocation/result");
+        then.status(200).json_body_obj(&qryd_job_result_completed);
+    });
 
-#[test]
-fn test_run_measuremt_registers() {
-    if env::var("QRYD_API_TOKEN").is_ok() {
-        pyo3::prepare_freethreaded_python();
-        Python::with_gil(|py| {
-            let backend = create_valid_backend_with_square_device(py, Some(11));
-            let program = create_quantum_program(true);
-            let failed_result = backend.call_method1("run_measurement_registers", (3_u32,));
-            assert!(failed_result.is_err());
-            let measurement = program.measurement();
-            let result: Registers = backend
-                .call_method1("run_measurement_registers", (measurement,))
+    pyo3::prepare_freethreaded_python();
+    Python::with_gil(|py| {
+        let backend =
+            create_valid_backend_with_square_device_mocked(py, Some(11), server.port().to_string());
```

Comment:
We should slighlty modify this. Great work on putting in the mocked tests, but it would be good if we still had the option to run the actual integration tests when a local ACCESS_TOKEN is detected in QRYD_API_TOKEN. Could you move the original if-statement in the test here and create two paths: one where the mock backend is created when the token is not in the environment and one where the non-mocked version is used?

That also goes for the other tests in here.